File: server/schemas/notification.py
# -*- coding: utf-8 -*-
"""알림 API 스키마 — NTF-001 Phase1."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class NotificationItem(BaseModel):
    noti_id: str
    farm_cd: str
    noti_type_cd: str
    noti_type_nm: str = ""
    priority_cd: str
    priority_nm: str = ""
    title: str
    body: str | None = None
    payload: dict[str, Any] | None = None
    source_cd: str
    ref_type: str | None = None
    ref_id: str | None = None
    event_at: str
    read_yn: str = "N"
    read_dt: str | None = None


class NotificationSummary(BaseModel):
    unread_count: int = 0
    urgent_count: int = 0


class NotificationReadResponse(BaseModel):
    noti_id: str
    read_yn: str = "Y"
    read_dt: str | None = None


class NotificationReadAllResponse(BaseModel):
    updated_count: int = Field(0, ge=0)

File: mobile/src/types/notification.ts
/** 알림 API — NTF-001 */

export type NotificationPayload = {
  /** NTF-001 SSOT 딥링크 키 */
  route?: string
  obs_id?: string
  work_dt?: string
  /** 지시서 호환 */
  route_name?: string
  params?: Record<string, string | number>
  path?: string
  obsId?: string
  workDt?: string
  [key: string]: unknown
}

export type NotificationItem = {
  noti_id: string
  farm_cd: string
  noti_type_cd: string
  noti_type_nm: string
  priority_cd: string
  priority_nm: string
  title: string
  body?: string | null
  payload?: NotificationPayload | null
  source_cd: string
  ref_type?: string | null
  ref_id?: string | null
  event_at: string
  read_yn: string
  read_dt?: string | null
}

export type NotificationSummary = {
  unread_count: number
  urgent_count: number
}

export type NotificationReadResponse = {
  noti_id: string
  read_yn: string
  read_dt?: string | null
}

export type NotificationReadAllResponse = {
  updated_count: number
}
